### ISSUE TITLE
[ts] Complete callback not added if a looping animation has a duration of 1 and deltaTime is an integer (a whole number)

### ISSUE THREAD

**Issue by Jeaper** [EsotericSoftware/spine-runtimes#2472]
if all three of these conditions are met, the complete callback will never be added to the animation

1. delta time is an integer (a whole number such as 17) (reachable with extreme timeScales, such as 1000) 
2. animation is looping
3. animation has a duration of 1

In AnimationState.ts   queueEvents() function 
```

// Duration is 1
// in this case trackLast will be an integer, causing trackLastWrapped to be 0
let trackLastWrapped = entry.trackLast % duration;
//....
//....
// in this case trackTime will be an integer, causing this if to be : (0 > 0) which always fails
if (entry.loop)
     complete = duration == 0 || trackLastWrapped > entry.trackTime % duration;
```

this was found in our case, when setting our timeScale to 1000 in order to skip animations, making our delta time not a floating number

**Comment by badlogic** [comment]
I don't see a way around this mathematically, so maybe we need to attack this from another angle.

You say you want to "skip" an animation. I assume you meant skip forward inside an animation that is currently being applied? If the duration is 1, and the animation is looping, why would you set `timeScale` to `1000` with a delta of `17`? Could you explain your use case a little or even better, provide a small example project that demonstrates what you are doing?

**Comment by NathanSweet** [comment]
To skip an animation, it would be better to set `trackTime` very high. That advances the time directly.

We detect the `complete` event by comparing the wrapped times to know that the current frame has looped as compared to the last frame. That means we need 2 frames to detect the "edge" of the complete event. If both frames fall on the modulo of the duration, the current code cannot detect the event.

By comparing the unwrapped times we can avoid this problem. Fixed in 88b009cd1fb14da1cc4e1d6438ad7888ee476462!
